cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -217,13 +223,21 @@ private static boolean subnetExists(List<String> subnets, String subnet) {
 
   // subnets are identified as
   // "https://www.googleapis.com/compute/v1/projects/<project>/regions/<region>/subnetworks/<name>"
-  // a subnet in the same region as the dataproc cluster must be chosen
-  private static String chooseSubnet(String network, List<String> subnets, String zone) {
+  // a subnet in the same region as the dataproc cluster must be chosen. If a subnet name is provided then the subnet
+  // will be choosen and the region will be picked on basis of the given zone. If a subnet name is not provided then
+  // any subnetwork in the region of the given zone will be picked.
+  private static String chooseSubnet(String network, List<String> subnets, @Nullable String subnet, String zone) {
```

Comment:
Not a part of this PR but any idea in which form we accept the subnet from the UI? Subnets are identified as `https://www.googleapis.com/compute/v1/projects/<project>/regions/<region>/subnetworks/<name>` so if we are accepting `<name>` part only then it's not guaranteed to be unique right?

Reply:
We are just accepting name.  
Given the name we pick the subnet which is regional (unique in a region). so from the zone we find the region to pick.